src/test/java/net/sf/jsqlparser/statement/alter/AlterTest.java

```diff
@@ -383,22 +407,38 @@ public void testAlterTableIndex586() throws Exception {
     public void testIssue259() throws JSQLParserException {
         assertSqlCanBeParsedAndDeparsed("ALTER TABLE feature_v2 ADD COLUMN third_user_id int (10) unsigned DEFAULT '0' COMMENT '第三方用户id' after kdt_id");
     }
-    
+
     @Test
     public void testIssue633_2() throws JSQLParserException {
-        assertSqlCanBeParsedAndDeparsed("CREATE INDEX idx_american_football_action_plays_1 ON american_football_action_plays USING btree (play_type)");
+        String statement = "CREATE INDEX idx_american_football_action_plays_1 ON american_football_action_plays USING btree (play_type)";
+        Statement parsed = assertSqlCanBeParsedAndDeparsed(statement);
+        CreateIndex created = new CreateIndex()
+                .withTable(new Table("american_football_action_plays"))
+                .withIndex(
+                        new Index().withName("idx_american_football_action_plays_1")
+                        .addColumns(new ColumnParams("play_type", null)).withUsing("btree")
+                        );
+        assertDeparse(created, statement);
+        assertEqualsObjectTree(parsed, created);
     }
-    
+
     @Test
     public void testAlterOnlyIssue928() throws JSQLParserException {
-        assertSqlCanBeParsedAndDeparsed("ALTER TABLE ONLY categories ADD CONSTRAINT pk_categories PRIMARY KEY (category_id)");
+        String statement = "ALTER TABLE ONLY categories ADD CONSTRAINT pk_categories PRIMARY KEY (category_id)";
```

Comment:
Don't replace existing tests. Create new tests.

Reply:
normally I would agree to the principle of not changing tests, but did'nt change the assert, but extending it with additional asserts.

Reply:
But for instance you changed on various places the getExpression() to getExpression(Classtype).

Reply:
For the rest you wrote I agree.

Reply:
getExpression(Classtype) calls getExpression() inside. I changed it if explicit casting was done within testcase.